// dash-button/build/__mocks__/console.js
'use strict';

var _stream = require('stream');

const ActualConsole = require.requireActual('console').Console;

class Console {

  constructor() {
    this.log = jest.fn((...args) => this._silentConsole.log(...args));
    this.info = jest.fn((...args) => this._silentConsole.info(...args));
    this.warn = jest.fn((...args) => this._silentConsole.warn(...args));
    this.error = jest.fn((...args) => this._silentConsole.error(...args));
    this.assert = jest.fn((...args) => this._silentConsole.assert(...args));
    this.dir = jest.fn((...args) => this._silentConsole.dir(...args));
    this.trace = jest.fn((...args) => this._silentConsole.trace(...args));
    this.time = jest.fn((...args) => this._silentConsole.time(...args));
    this.timeEnd = jest.fn((...args) => this._silentConsole.timeEnd(...args));

    this._silentConsole = new ActualConsole(new NullWritableStream());
  }

}

Console.Console = Console;
class NullWritableStream extends _stream.Writable {
  _write(chunk, encoding, callback) {
    callback();
  }
}

module.exports = new Console();
